File: bindings/java-jni/src/main/java/org/maplibre/nativejni/map/DebugOption.java
package org.maplibre.nativejni.map;

/** Map debug overlay options. */
public enum DebugOption {
  TILE_BORDERS(1 << 1),
  PARSE_STATUS(1 << 2),
  TIMESTAMPS(1 << 3),
  COLLISION(1 << 4),
  OVERDRAW(1 << 5),
  STENCIL_CLIP(1 << 6),
  DEPTH_BUFFER(1 << 7);

  private final int nativeMask;

  DebugOption(int nativeMask) {
    this.nativeMask = nativeMask;
  }

  public int nativeMask() {
    return nativeMask;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/map/DebugOption.kt
package org.maplibre.nativeffi.map

/** Map debug overlay options. */
public enum class DebugOption(internal val nativeMask: UInt) {
  TILE_BORDERS(1U shl 1),
  PARSE_STATUS(1U shl 2),
  TIMESTAMPS(1U shl 3),
  COLLISION(1U shl 4),
  OVERDRAW(1U shl 5),
  STENCIL_CLIP(1U shl 6),
  DEPTH_BUFFER(1U shl 7),
}
